dadi/lib/auth/index.js

```diff
@@ -1,184 +1,129 @@
-
-  // all /config requests must be authenticated
-  if (parsedUrl.pathname.indexOf('config') > -1) return true
-
-  // /hello welcome route doesn't require authentication
-  if (req.url === '/hello') return false
-
-  // docs requests don't need to be authenticated
-  if (parsedUrl.pathname.indexOf('docs') > 0) return false
-
-  const endpointKey = _.find(_.keys(endpoints), function (k) {
-    return parsedUrl.pathname.match(pathToRegexp(k))
+  server.routeMethods(this.tokenRoute, {
+    post: this.generateToken.bind(this)
   })
 
-  if (!endpointKey) return true
-
-  const endpoint = endpoints[endpointKey]
-
-  if (isMediaEndpoint(endpoint, req)) {
-    if (req.params.token && req.params.token === 'sign') {
-      return true
-    } else {
-      // no auth required, token is an actual token or req.params.filename exists
-      return false
-    }
-  }
-
-  if (endpoint.model && endpoint.model.settings && endpoint.model.settings.hasOwnProperty('authenticate')) {
-    if (typeof endpoint.model.settings.authenticate === 'boolean') {
-      return endpoint.model.settings.authenticate
-    } else {
-      return endpoint.model.settings.authenticate.indexOf(req.method) > -1
-    }
-  } else {
-    return true
-  }
+  server.use(
+    this.authenticateRequest.bind(this)
+  )
 }
 
 /**
- * If the URL is for a media asset, but not at the root, then it can be unauthenticated
- * @param {Object} endpoint - the component/controller matching the request URL
- * @param {Object} req - the original HTTP request
- * @returns {Boolean} - returns true if the URL starts with '/api/media/' followed by a filename
+ * Intercepts every request, apart from the ones targeting
+ * the token store URL, and checks for the presence of a
+ * bearer token in the appropriate header. If one is found,
+ * it's JWT-decoded and in case of success, the resulting
+ * client is added to the request as `req.dadiApiClient`.
+ *
+ * @param  {http.IncomingMessage}   req
+ * @param  {http.ServerResponse}    res
+ * @param  {Function}               next
  */
-function isMediaEndpoint (endpoint, req) {
-  return endpoint.model &&
-    endpoint.model.settings &&
-    endpoint.model.settings.type &&
-    endpoint.model.settings.type === 'mediaCollection'
-}
+AuthMiddleware.prototype.authenticateRequest = function (req, res, next) {
+  if (req.url === this.tokenRoute) return next()
 
-function isAuthorized (endpoints, req, client) {
-  const path = url.parse(req.url, true)
+  let header = req.headers && req.headers.authorization
+  let match = header && header.match(/^Bearer (.*)$/)
 
-  const urlParts = _.compact(path.pathname.split('/'))
-  const version = urlParts.shift()
+  if (!match) return next()
 
-  const endpointKey = _.find(_.keys(endpoints), function (k) { return k.indexOf(path.pathname) > -1 })
+  jwt.verify(
+    match[1],
+    config.get('auth.tokenKey'),
+    (err, decoded) => {
+      if (err) {
+        req.dadiApiClientError = err
```

Comment:
I'm not sure about having a different property in the case of an error - what about attaching errors to req.dadiApiClient instead so there's only one additional property added to the request (I'm thinking from a debugging perspective more than anything)

Reply:
I'm happy with that! Will change.